service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/gcp/ainotebook/post-startup.sh

```diff
@@ -756,6 +756,32 @@ fi
 chown ${LOGIN_USER}:${LOGIN_USER} "${USER_BASHRC}"
 chown ${LOGIN_USER}:${LOGIN_USER} "${USER_BASH_PROFILE}"
 
+#######################################
+# Restart proxy to pick up the new env
+######################################
+APP_PROXY=$(get_metadata_value "instance/attributes/terra-app-proxy")
+readonly PROXY_ENV="/opt/deeplearning/proxy.env"
+if [[ -n "${APP_PROXY}" ]]; then
+    echo "Using custom Proxy Agent"
+    RESOURCE_ID=$(get_metadata_value "instance/attributes/terra-resource-id")
+    NEW_PROXY="https://${APP_PROXY}"
+    ESCAPED_NEW_PROXY=$(sed 's/[\/&]/\\&/g' <<< "${NEW_PROXY}")
+    NEW_PROXY_URL="${RESOURCE_ID}.${APP_PROXY}"
+    ESCAPED_NEW_PROXY_URL=$(sed 's/[\/&]/\\&/g' <<< "${NEW_PROXY_URL}")
+    sed -i "s/^PROXY_REGISTRATION_URL=.*/PROXY_REGISTRATION_URL=${ESCAPED_NEW_PROXY}/" "${PROXY_ENV}"
+    sed -i "s/^PROXY_URL=.*/PROXY_URL=${ESCAPED_NEW_PROXY_URL}/" "${PROXY_ENV}"
+    sed -i "s/^BACKEND_ID=.*/BACKEND_ID=${RESOURCE_ID}/" "${PROXY_ENV}"
+    systemctl restart notebooks-proxy-agent.service
+    echo "Proxy Agent service restarted"
+    INSTANCE_NAME=$(get_metadata_value "instance/name")
+    INSTANCE_ZONE="/"$(get_metadata_value "instance/zone")
+    INSTANCE_ZONE="${INSTANCE_ZONE##/*/}"
+    timeout 30 gcloud compute instances add-metadata "${INSTANCE_NAME}" \
```

Comment:
Setting metadata can take a little while, but did you specifically run into an issue prompting adding the timeout? There should be some commentary about the choice in adding the timeout.

Reply:
no this was not from my local testing. i copied this straight from the deeplearning proxy setup script. https://source.corp.google.com/piper///depot/google3/cloud/ml/dset/dlenv/build/vm/packer/base/proxy_agent/attempt-register-vm-on-proxy.sh;l=286

Reply:
Interesting, should we borrow their comment in some form as well?

Reply:
i don't think that comment is related to the timeout. it's talking about execution order. i also can't see the content of the bug /shrug